geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/QueryPlanFilterVisitor.scala

```diff
@@ -316,6 +318,16 @@ class QueryPlanFilterVisitor(sft: SimpleFeatureType) extends DuplicatingFilterVi
     }
   }
 
+  override def visit(filter: PropertyIsLike, extraData: Any): AnyRef = {
+    try {
+      new LikeToRegexConverter(filter).getPattern
+    } catch {
+      case e: Exception =>
+        throw new IllegalArgumentException(s"The regex filter (${ECQL.toCQL(filter)}) for the (i)like filter is invalid.", e)
```

Comment:
is there any possibility that ECQL.toCQL() might throw an exception here as well?

Reply:
you should be able to use `filter.getLiteral` to return the raw string value, which would be safe

Reply:
Ok, switched to filter.getLiteral.